### DISCUSSION TITLE
CrossOverUp and CrossOverDown indicators

### DISCUSSION THREAD

**Discussion by lakkakula-shashidhar-rao** [DaveSkender/Stock.Indicators#1189] [Q&A] [answered]
there are many strategies that need to take position on the crossover of any indicator, for example, simple two ema  cross over EMA5 and EMA10, when EMA5 crosses over EMA10 to Up side, then take long position and when EMA5 CrossesOver to down side, we take a short position.

Having an indicator which takes two results or one result and one param of quote (open / high / low/ close) so that we can know the places where the crossover happened in the entire quote series would be great.

Tried giving you a reference of what is available in TA4J - Technical Analysis Library for Java.

![image](https://github.com/DaveSkender/Stock.Indicators/assets/61023532/4e57989c-9fe5-476d-9c13-3b146885939a)

**Comment by DaveSkender** [comment, marked as answer]
In order to maintain a [single responsibility](https://github.com/DaveSkender/Stock.Indicators/discussions/648) of this library we’re limiting its features to indicators only.  Definition of trade triggering events, like crossovers, would be more appropriate for an algo-trading library or app.

**Reply by lakkakula-shashidhar-rao** [reply]
Thank your for the response, and understand your perspective, appreciate it.

would you please be able to help or point me to some code on how to get this cross over

**Reply by DaveSkender** [reply]
I don't have any code available, but if you're iterating on RSI for example, and you wanted to know when it crosses 70, it'd be something like:

```csharp
for(int i=1; i<total; i++){
  bool crossover = rsi[i-1] < 70 && rsi[i] >= 70;
  bool crossunder = rsi[i-1] > 70 && rsi[i] <= 70;
}
```

**Reply by lakkakula-shashidhar-rao** [reply]
thanks for the code snippet
